### DISCUSSION TITLE
Style the Home idp page

### DISCUSSION THREAD

**Discussion by nikitadolina** [sventorben/keycloak-home-idp-discovery#153] [Q&A] [answered]
**Edit**:
Found the answer after all, it's called LoginUsername.

**Init question**
Im styling the login flow with a react helper called Keycloakify.
But the Home-idp-discorvey login page is not triggering the login.ftl, do you know what page component it loads (in keycloakify terms)?

**Comment by pcMarcos** [comment]
I'm having the same issue. I'm trying to add more html elements to the login.ftl but the idp page is not reading my file. I think it inherits from base but it's not being possible to change it.

**Reply by sventorben** [reply, marked as answer]
Please try the login-username.ftl instead.

**Reply by pcMarcos** [reply]
it worked perfectly. after applying the `login-username.ftl` and `login-password.ftl` to style the entire flow. 

thanks a lot @sventorben

**Comment by ryancamp1** [comment]
@sventorben I have updated these files login-username.ftl and login-password.ftl, however I cannot seem to style this injected email box
![image](https://github.com/user-attachments/assets/81701d71-5182-4d2c-9995-b049e524c211)

This only seems to happen on the first login page, on load, if I then make use of home-idp plugin and it re-directs to the password page this is styled as per my theme.
Any ideas?

**Reply by sventorben** [reply]
Sorry, no idea. The home-idp extension only makes use of the `login-username.ftl` template by calling `createLoginUsername()` on the `FreeMarkerLoginFormsProvider`.
